File: After BootCamp/week6/783-minimum-distance-between-bst-nodes/783-minimum-distance-between-bst-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDiffInBST(self, root: Optional[TreeNode]) -> int:
        
        ar = self.tr(root)
        
        res = float("inf")
        
        for i in range(1,len(ar)):
            res = min(res,ar[i] - ar[i-1])
        
        return res
    
    
    def tr(self,node):
        if not node:
            return []
        
        return self.tr(node.left) + [node.val] + self.tr(node.right)

File: Project Phase Camp/783-minimum-distance-between-bst-nodes/783-minimum-distance-between-bst-nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function(root) {
    
    const walk = (node) => {
        if(node === null){
            return [];
        }
        
        const left = walk(node.left);
        const right = walk(node.right);
        
        let temp = left.concat([node.val]);
        let cur = temp.concat(right);
        
        return cur;
    };
    
    let ar = walk(root);
    let ans = Infinity;
    console.log(ar);
    for(let i = 0; i < ar.length - 1; i++){
        ans = Math.min(ar[i + 1] - ar[i], ans);
    }
    
    return ans;
};